src/ACEtk/details/verify/src/positive.hpp

```diff
@@ -0,0 +1,27 @@
+template< typename Range >
+inline bool positive( Range& array ){
+  auto firstNegative = ranges::count_if( array, 
+    [](auto& e ){ return e < 0.0; } );
+
+  if( firstNegative > 0 ){
+    njoy::Log::error( "All elements of the array must be positive" );
+    njoy::Log::info( "Found {} negative array elements", firstNegative );
+    return false;
+  }
+
+  return true;
+}
+
+template< typename Range >
+inline bool strictlyPositive( Range& array ){
```

Comment:
the conventions for positive and strictly positive are backwards

Reply:
I think you are incorrect. [Strictly positive](https://en.wikipedia.org/wiki/Strictly_positive_measure) ([also](https://proofwiki.org/wiki/Definition:Strictly_Positive/Number)) means that it is "nowhere zero" so I have counted the number of times that it is `<=`. If that count is more than `0`, then I know I've found (at least) one that fails.